apps/zk-explorer/components/RadioCard.tsx

```diff
@@ -0,0 +1,79 @@
+import { InfoCircledIcon } from '@radix-ui/react-icons';
+import { IconProps } from '@radix-ui/react-icons/dist/types';
+import { Typography } from '@webb-tools/webb-ui-components';
+import { ComponentType, FC, useCallback } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+export type RadioCardProps = {
+  id: string;
+  title: string;
+  tooltip?: string;
+  description?: string;
+  Icon?: ComponentType<IconProps>;
+  selectedRadioItem: string | null;
+  setSelectedRadioItem: (id: string) => void;
+};
+
+export const RadioCard: FC<RadioCardProps> = ({
+  id,
+  setSelectedRadioItem,
+  selectedRadioItem,
+  title,
+  tooltip,
+  description,
+  Icon,
+}) => {
+  const isChecked = selectedRadioItem === id;
```

Comment:
Add useMemo

Reply:
The comparison is a constant time operation, so adding `useMemo` here might introduce overhead. Should I still proceed with using it?

Reply:
I don't really understand the **constant time operation** part, can you elaborate more on that?

Reply:
My thought process was that comparisons (say equality comparisons) in JavaScript and most languages are fast and efficient operations (aka O(1) worst-case).

So, using `useMemo` (whose primary purpose is to optimize by memoizing complex, computationally heavy operations) would not provide a benefit and perhaps even add overhead (function call vs. simple comparison operation).

Every time the component is re-rendered, the variable is re-computed. But that wouldn't mean that anything depending on that variable would *re-render* as well since React only re-renders when the variable's value *actually* changes.

For operations that process arrays linearly (like `for every element in X, do Y`), more complex data structures, or computationally heavy calculations, `useMemo` would be more appropriate.